### PR TITLE
Fixes issue #69, Add pre-commit hooks for linting 

### LINKED ISSUE #69: Add pre-commit hooks for linting
Add pre-commit hooks that runs linting before committing the file.

### Requirements:
 - [ ] Add `.pre-commit-config.yml` and define eslint and prettier hooks
 - [ ] Update the docs to install pre-commit hooks

### Reference: 
- https://pre-commit.com/

### PR DESCRIPTION
**Pull request by rithgan**
# Related Issue
- closes #69 

# Proposed Changes
- Added pre-commit config for prettier and eslint
- Update the docs accordingly

# Additional Info
- There is a file, src/public/scripts/script.js, which requires review. pre-commit hooks reported 32 problems with it. I did not touch it as it may be important. 
Suggestion - A separate issue can be created of level 1 to solve it. 

# Checklist
- [x] Tests
- [x] Translations
- [x] Documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,41 +37,74 @@ Try it out <a href="https://rest-fetch.herokuapp.com/)"><img src="https://github
...
+
+> $ brew install pre-commit
+
+-   Using conda (via conda-forge):
+
+> $ conda install -c conda-forge pre-commit
+
+After completing the above steps, verify your installation by running
+
+> $ pre-commit --version
```
Comment (ShauryaAg) [inline README.md:59]: Make the code snippets as `code`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+repos:
+    - repo: https://github.com/pre-commit/mirrors-prettier
+      rev: "v2.2.1" # Use the sha or tag you want to point at
+      hooks:
+          - id: prettier
+    - repo: https://github.com/pre-commit/mirrors-eslint
+      rev: "v7.23.0" # Use the sha / tag you want to point at
+      hooks:
+          - id: eslint
```
Comment (ShauryaAg) [inline .pre-commit-config.yaml:9]: Specify the types (file extensions) we need to lint.
Reference: [eslint](https://github.com/pre-commit/mirrors-eslint), [prettier](https://github.com/pre-commit/mirrors-prettier)

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (rithgan) [reply .pre-commit-config.yaml:9]: I have specified file extension for eslint, but I think prettier is supposed to run on all files.

Comment (ShauryaAg) [reply .pre-commit-config.yaml:9]: You can specify types in prettier as

```yaml
...
  -   id: prettier
      types: [javascript]
...
```

Comment (ShauryaAg) [reply .pre-commit-config.yaml:9]: Please also specify `--fix` as `args` in eslint hook

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (rithgan) [reply .pre-commit-config.yaml:9]: By default, it would run on all files. If you still need me to specify just for js and ts, I can do that.

Comment (ShauryaAg) [reply .pre-commit-config.yaml:9]: Please specify it run on only `js` and `ts` files.

Comment (rithgan) [reply .pre-commit-config.yaml:10]: Please review.

**Review by ShauryaAg — APPROVED** [review]

**Pull request merged**